### DISCUSSION TITLE
problem with tmux characters

### DISCUSSION THREAD

**Discussion by minaaaayk** [sourcelair/xterm.js#4511] [General]
I connect the xtrem.js and tmux using webSocket. and I have a problem with showing some characters like line splitter like this:
![image](https://github.com/xtermjs/xterm.js/assets/37193272/445fb4df-6515-4ff4-becd-4c3a4f9a4389)

## Details
- Browser and browser version: Version 112.0.5615.165 (Official Build) (64-bit)
- OS version: Ubuntu 22.04 LTS 
- xterm.js version: 5.1.0

**Comment by minaaaayk** [comment]
this is my xterm configs:
![image](https://github.com/xtermjs/xterm.js/assets/37193272/c9dac6ca-1b8c-4377-bd91-39387baaeeaa)

**Comment by jerch** [comment]
This is an encoding issue between the tmux data and xterm.js, where utf-8 byte values are wrongly treated as JS string. How do you forward the data to xterm.js?

**Reply by minaaaayk** [reply]
of course. I used `window.btoa()` to convert data.  and finally found this link [https://blog.linwenguo.com/archives/53/](https://blog.linwenguo.com/archives/53/):

```
const commonContent = msg.replace(/\s/g, '+');
const data = Buffer.from(commonContent, 'base64').toString();
this.terminal.write(data);
```

**Reply by jerch** [reply]
Well `Buffer` is normally not available on the browser. For a more generic browser-ish way, see https://github.com/xtermjs/xterm.js/discussions/4481#discussioncomment-5657455.
